language-plutus-core/src/Language/PlutusCore/Constant/Dynamic/Instances.hs

```diff
@@ -88,28 +88,28 @@ instance KnownDynamicBuiltinType Bool where
     makeDynamicBuiltin b = pure $ if b then true else false
 
     readDynamicBuiltin eval b = do
-        let int1 = TyApp () (TyBuiltin () TyInteger) (TyInt () 4)
-            asInt1 = Constant () . BuiltinInt () 1
+        let int = TyBuiltin () TyInteger
+            asInt = Constant () . BuiltinInt ()
         -- Encode 'Bool' from Haskell as @integer 1@ from PLC.
-        res <- eval mempty (mkIterApp () (TyInst () b int1) [asInt1 1, asInt1 0])
+        res <- eval mempty (mkIterApp () (TyInst () b int) [asInt 1, asInt 0])
         pure $ lift res >>= \case
-            Constant () (BuiltinInt () 1 1) -> pure True
-            Constant () (BuiltinInt () 1 0) -> pure False
-            _                               -> throwError "Not an integer-encoded Bool"
+            Constant () (BuiltinInt () 1) -> pure True
+            Constant () (BuiltinInt () 0) -> pure False
+            _                             -> throwError "Not an integer-encoded Bool"
 
 -- Encode 'Char' from Haskell as @integer 4@ from PLC.
 instance KnownDynamicBuiltinType Char where
-    toTypeEncoding _ = TyApp () (TyBuiltin () TyInteger) (TyInt () 4)
+    toTypeEncoding _ = TyBuiltin () TyInteger
 
-    makeDynamicBuiltin = fmap (Constant ()) . makeBuiltinInt 4 . fromIntegral . ord
+    makeDynamicBuiltin = Just . Constant () . makeBuiltinInt . fromIntegral . ord
```

Comment:
I wonder whether there are now any cases where we want to return `Nothing` rather than `Just`. I guess, I'll try to remove the `Maybe` later.